ql/src/test/queries/clientpositive/udf_configurable.q

```diff
@@ -0,0 +1,55 @@
+CREATE TABLE test_udf_configurable (cint1 INT, cint2 INT, ctimestamp TIMESTAMP, text_timestamp STRING);
+INSERT INTO test_udf_configurable VALUES
+  (10000, 3, CAST('1970-01-01 01:02:03' AS TIMESTAMP), '1970-01-01 01:02:03 4'),
+  (20000, 5, CAST('1970-01-02 04:05:06' AS TIMESTAMP), '1970-01-02 04:05:06 5'),
+  (30000, 7, CAST('1970-01-03 07:08:09' AS TIMESTAMP), '1970-01-03 07:08:09 6');
+
+set hive.compat=latest;
+set hive.strict.timestamp.conversion=false;
+set hive.int.timestamp.conversion.in.seconds=true;
+set hive.local.time.zone=Asia/Bangkok;
+set hive.datetime.formatter=SIMPLE;
+set hive.masking.algo=sha512;
+set hive.use.googleregex.engine=true;
+
+-- On HiveServer2
+SELECT
+  -- DECIMAL because of hive.compat=latest
+  TYPEOF(cint1 / cint2),
```

Comment:
nit: this `TYPEOF(cint1 / cint2)` happens only at compile stage, should we change it to `cint1 / cint2`?

Reply:
You are right! I wonder what is the most reasonable way to identify if it is DOUBLE or DECIMAL...

Reply:
As you say, I found we can simply remove TYPEOF since the text expression or precision of DOUBLE and DECIMAL is different. This is the diff between `hive.compat=latest` vs default.

```
< 3333.3333333333335	false	1970-01-01 02:46:40	1970-01-01 01:02:03 4	1970-01-01 09:46:40 4	-21477	-21477	ca764b57c635a893c91b0edaea84bca142e065990e1a66b9c60cf855777af51070444210d30e359fa81e5a77c68a073224e0c7343d957f556816618a10baa37c	false
< 4000.0	false	1970-01-01 05:33:20	1970-01-02 04:05:06 5	1970-01-01 12:33:20 4	75906	75906	7882085a4ed36e7c675fcd12083eefa9a208f1fdcfba10b3a986bb5b1d43e23da3ba87b3933feebfed80ba9e05e4e73f72231cb8022453e562e034e8b5c603b9	false
< 4285.714285714285	false	1970-01-01 08:20:00	1970-01-03 07:08:09 6	1970-01-01 15:20:00 4	173289	173289	d9899e12fce35a4f65cd63aee1e953dacd9e91229d52dd17ddad8e44c17d2d081c0a5519a96637663bc6baa2d7c96d6f1a9d6944ac7668c7926ebc10c9f3b81b	false
---
> 3333.33333333333	false	1970-01-01 02:46:40	1970-01-01 01:02:03 4	1970-01-01 09:46:40 4	-21477	-21477	ca764b57c635a893c91b0edaea84bca142e065990e1a66b9c60cf855777af51070444210d30e359fa81e5a77c68a073224e0c7343d957f556816618a10baa37c	false
> 4000.00000000000	false	1970-01-01 05:33:20	1970-01-02 04:05:06 5	1970-01-01 12:33:20 4	75906	75906	7882085a4ed36e7c675fcd12083eefa9a208f1fdcfba10b3a986bb5b1d43e23da3ba87b3933feebfed80ba9e05e4e73f72231cb8022453e562e034e8b5c603b9	false
> 4285.71428571429	false	1970-01-01 08:20:00	1970-01-03 07:08:09 6	1970-01-01 15:20:00 4	173289	173289	d9899e12fce35a4f65cd63aee1e953dacd9e91229d52dd17ddad8e44c17d2d081c0a5519a96637663bc6baa2d7c96d6f1a9d6944ac7668c7926ebc10c9f3b81b	false
```

Reply:
I pushed the change and am waiting for CI.
https://github.com/apache/hive/pull/5278/commits/db8edc83ae2e0ba44c607196c61ed7e60c513561